src/Uno.UWP/UI/Notifications/ToastNotifier.Android.cs

```diff
@@ -0,0 +1,199 @@
+				iconId = resolveInfo.IconResource;
+			}
+
+			builder.SetSmallIcon(iconId);
+
+
+			// for somewhat older Android, we need to set Priority (deprecated in API26)
+
+			if (Android.OS.Build.VERSION.SdkInt < Android.OS.BuildVersionCodes.O)
+			{
+#pragma warning disable CS0618 // Type or member is obsolete
+				builder.SetPriority(0); // PRIORITY_DEFAULT
+#pragma warning restore CS0618 // Type or member is obsolete
+			}
+
+
+			// mirroring / bridging of notification
+			builder.SetLocalOnly((notification.NotificationMirroring == NotificationMirroring.Disabled));
+
+			// expiration time
+			if((notification.ExpirationTime != null) && (notification.ExpirationTime.HasValue) )
+			{
+				int seconds = (int)(notification.ExpirationTime.Value - DateTime.Now).TotalSeconds;
+				if(seconds>0)
+				{
+					builder.SetTimeoutAfter(seconds * 1000);
+				}
+
+			}
+
+
+			// action - opening app from tap on Toast (App.OnActivated)
+
+			var intentAction = Android.App.Application.Context.PackageManager.GetLaunchIntentForPackage(packageName);
+			intentAction.SetPackage(packageName);
+			// according to doc:
+			// The name must include a package prefix, for example the app com.android.contacts would use names like "com.android.contacts.ShowAll"
+			// but it works also in this way, and getting package name in background app can be difficult.
+			intentAction.PutExtra("Uno.internal.IntentType", (double)ApplicationModel.Activation.ActivationKind.ToastNotification);
+
+			string toastArgument = "";
+			var xmlDoc = new XmlDocument();
+			try
+			{
+				xmlDoc.LoadXml(notification.Content.GetXml());  // Windows.Data.Xml.Dom is almost not implemented, so we have to use System.Xml
+				var childToast = xmlDoc.GetElementsByTagName("toast");
+				if(childToast.Count == 1)
+				{
+					var childLaunch = childToast.Item(0).Attributes.GetNamedItem("launch");
+					if(childLaunch != null)
+					{
+						toastArgument = childLaunch.Value;
+					}
+				}
+			}
+			catch
+			{
+				// error parsing XML - maybe simple not 'launch' attribute
+				toastArgument = "";
+			}
+			intentAction.PutExtra("Uno.internal.ToastArgument", toastArgument);
+
+			// we should have intent.data field different for different toast arguments,
+			// else - next intent would overwrite previous.
+			var data = Android.Net.Uri.Parse("http://unotask.com/toast?arg=" + Android.Net.Uri.Encode(toastArgument));
+			intentAction.SetData(data);
+
+			var pendingIntent = Android.App.PendingIntent.GetActivity(Android.App.Application.Context, 12345, intentAction, 0); // 12345=requestCode
+			builder.SetContentIntent(pendingIntent);
+
+			builder.SetAutoCancel(true);
+
+
+			// another 'must' - uniq Tag/id pair
+
+			string notifTag = notification.Tag;
+			if (string.IsNullOrEmpty(notifTag))
+			{
+				var dateForTag = DateTime.UtcNow.AddYears(-2010);   // make number smaller (but cannot use 2020! "The added or subtracted value results in an un-representable DateTime.\nParameter name: months")

```

Comment:
Can new guid be used instead ?

Reply:
Yes. Is Guid really better?

Reply:
Yes, it's not explicitly relying on transient state like time.